network-config-analyzer/IstioTrafficResourcesYamlParser.py

```diff
@@ -51,17 +51,15 @@ def parse_gateway(self, gateway_resource, gateway_file_name):
         :param str gateway_file_name: the name of the gateway resource file (for reporting errors and warnings)
         """
         self.set_file_name(gateway_file_name)  # for error/warning messages
-        self.check_fields_validity(gateway_resource, 'Gateway',
-                                   {'apiVersion': [1, str], 'kind': [1, str],
-                                    'metadata': [1, dict], 'spec': [1, dict]})
-        if gateway_resource['kind'] != 'Gateway' or 'networking.istio.io' not in gateway_resource['apiVersion']:
-            return  # Not an Istio Gateway object
-        metadata = gateway_resource['metadata']
-        self.check_fields_validity(metadata, 'Gateway metadata', {'name': [1, str], 'namespace': [0, str]})
-        gtw_name = metadata['name']
-        gtw_namespace = self.peer_container.get_namespace(metadata.get('namespace', 'default'))
-        gtw_spec = gateway_resource['spec']
+        gtw_name, gtw_ns = self.parse_generic_yaml_objects_fields(gateway_resource, 'Gateway',
```

Comment:
The second parameter for `parse_generic_yaml_objects_fields `should be of type `list`, but here it is called with `str`.